src/utilities.jl

```diff
@@ -57,32 +57,25 @@ function normalized_variance_cumsum(A::AbstractMatrix)
     vars ./ last(vars)
 end
 
-const strictlowertrid = Dict(
-    1 => (),
-    2 => ((2,1),),
-    3 => ((2,1), (3,1), (3,2)),
-    4 => ((2,1), (3,1), (4,1), (3,2), (4,2), (4,3)),
-)
-
-function lowertriinds(k)
-    indpairs = NTuple{2,Int}[]
-    for j in 1:(k-1)
-        for i in (j+1):k
-            push!(indpairs, (i,j))
-        end
-    end
-    (indpairs...,)
-end
+"""
+    ltriindprs
 
+A row-major order `Vector{NTuple{2,Int}}` of indices in the strict lower triangle.
 """
-    indpairs(k)
 
-Return a tuple of (row,column) tuples in the strict lower triangle of a `k` by `k` matrix.
+const ltriindprs = NTuple{2,Int}[]
 
-The results are memoized in the `strictlowertrid` `Dict` and created by `lowertriinds(k)`
-when needed.
-"""
-indpairs(k) = get!(strictlowertrid, k, lowertriinds(k))
+function checkindprsk(k::Integer)
+    kchoose2 = (k * (k - 1)) >> 1
+    global ltriindprs
```

Comment:
why is this global?

Reply:
I found I needed to specify `global` when I was developing the function but that may have been because I was doing it in the REPL.  I'll see what happens if I delete this.  I do want to use the value declared in the module so that it is persistent.